projects/plutus/src/ltc_afe.c

```diff
@@ -0,0 +1,279 @@
+  // ADAX
+  uint16_t adax = LTC6804_ADAX_RESERVED | LTC6804_ADAX_GPIO1 | (mode << 7);
+
+  uint8_t cmd[4] = { 0 };
+  prv_build_cmd(adax, cmd, SIZEOF_ARRAY(cmd));
+
+  prv_wakeup_idle(afe);
+  spi_exchange(afe->spi_port, cmd, 4, NULL, 0);
+
+  // wait for conversions to finish
+  delay_ms(10);
+}
+
+// write config to all devices
+static StatusCode prv_write_config(const LtcAfeSettings *afe, uint8_t gpio_enable_pins) {
+  // see p.54 in datasheet
+  LtcAfeWriteConfigPacket config_packet = { 0 };
+
+  prv_build_cmd(LTC6804_WRCFG_RESERVED, config_packet.wrcfg, SIZEOF_ARRAY(config_packet.wrcfg));
+
+  // essentially, each set of CFGR registers are clocked through each device,
+  // until the first set reaches the last device (like a giant shift register)
+  // thus, we send CFGR registers starting with the bottom slave in the stack
+  for (uint8_t device = PLUTUS_AFE_DEVICES_IN_CHAIN; device > 0; --device) {
+    uint8_t curr_device = PLUTUS_AFE_DEVICES_IN_CHAIN - device;
+    uint8_t enable = gpio_enable_pins;
+    uint16_t undervoltage = 0;
+    uint16_t overvoltage = 0;
+
+    uint16_t cell_index = curr_device * LTC6804_CELLS_PER_DEVICE;
+    config_packet.devices[curr_device].reg.discharge_c1 = s_discharging_cells[cell_index + 0];
+    config_packet.devices[curr_device].reg.discharge_c2 = s_discharging_cells[cell_index + 1];
+    config_packet.devices[curr_device].reg.discharge_c3 = s_discharging_cells[cell_index + 2];
+    config_packet.devices[curr_device].reg.discharge_c4 = s_discharging_cells[cell_index + 3];
+    config_packet.devices[curr_device].reg.discharge_c5 = s_discharging_cells[cell_index + 4];
+    config_packet.devices[curr_device].reg.discharge_c6 = s_discharging_cells[cell_index + 5];
+    config_packet.devices[curr_device].reg.discharge_c7 = s_discharging_cells[cell_index + 6];
+    config_packet.devices[curr_device].reg.discharge_c8 = s_discharging_cells[cell_index + 7];
+    config_packet.devices[curr_device].reg.discharge_c9 = s_discharging_cells[cell_index + 8];
+    config_packet.devices[curr_device].reg.discharge_c10 = s_discharging_cells[cell_index + 9];
+    config_packet.devices[curr_device].reg.discharge_c11 = s_discharging_cells[cell_index + 10];
+    config_packet.devices[curr_device].reg.discharge_c12 = s_discharging_cells[cell_index + 11];
+
+    config_packet.devices[curr_device].reg.discharge_timeout = LTC_AFE_DISCHARGE_TIMEOUT_1_MIN;
+
+    config_packet.devices[curr_device].reg.adcopt = ((afe->adc_mode + 1) > 3);
+    config_packet.devices[curr_device].reg.swtrd = true;
+
+    config_packet.devices[curr_device].reg.undervoltage = undervoltage;
+    config_packet.devices[curr_device].reg.overvoltage = overvoltage;
+
+    // GPIO5, ..., GPIO2 are used to MUX data
+    config_packet.devices[curr_device].reg.gpio = (enable >> 3);
+
+    uint16_t cfgr_pec = crc15_calculate((uint8_t *)&config_packet.devices[curr_device].reg, 6);
+    config_packet.devices[curr_device].pec = SWAP_UINT16(cfgr_pec);
```

Comment:
Maybe add a comment explaining why you swap things around in general (big endian vs. little endian)